src/classes/CW_AudienceGroupCompControlerTest.cls

```diff
@@ -0,0 +1,270 @@
+			system.assertEquals(true  ,compCtrl.getOutputMode());
+			system.assertEquals(16    ,compCtrl.totalSize);
+			system.assertEquals(4     ,compCtrl.getTotalPages());
+			CompCtrl.addGroup();
+			compCtrl.newRec.Name='Test_Name';
+			compCtrl.newRec.Description__c = 'Test Descr';
+
+			compCtrl.SaveNewRecord();
+			//Chech Error Messages
+			system.assertEquals(true  ,ApexPages.hasMessages());
+			ApexPages.getMessages().clear();
+			compCtrl.newRec.Name='Test_Name2';
+			compCtrl.newRec.Is_Control_Group__c = false;
+			compCtrl.newRec.Control_Group__c=listGroups[1].Id;
+			compCtrl.Cancel();
+			
+			compCtrl.selectAll = true;
+			compCtrl.selectDeselectAllItems();
+			system.assertEquals(5     ,compCtrl.getSelectedCount());
+			system.assertEquals(false ,compCtrl.getDeleteDisabled());
+			
+			compCtrl.selectAll = false;
+			compCtrl.selectDeselectAllItems();
+			system.assertEquals(0     ,compCtrl.getSelectedCount());
+			system.assertEquals(true  ,compCtrl.getDeleteDisabled());
+			
+			compCtrl.selectedGroupIds.Add(listGroups[3].Id);
+			compCtrl.selectedGroupIds.Add(listGroups[4].Id);
+			system.assertEquals(false  ,compCtrl.getDeleteDisabled());
+			compCtrl.getDeleteMessage();
+			compCtrl.deleteGroups();
+			system.assertEquals(14    , RecList.Size());
+			system.assertEquals(5     ,compCtrl.campGroups.Size());
+			system.assertEquals(true  ,compCtrl.getOutputMode());
+			//Tray to delete used control groups
+			compCtrl.selectedGroupIds.Add(listGroups[0].Id);
+			compCtrl.selectedGroupIds.Add(listGroups[1].Id);
+			compCtrl.selectAll = false;
+			compCtrl.selectDeselectAllItems();
+			
+			compCtrl.getDeleteMessage();
+			compCtrl.deleteGroups();
+			
+			//check for errors
+			system.assertEquals(true  ,ApexPages.hasMessages());
+			system.assertEquals(14    , RecList.Size());
+			
+			compCtrl.contextItem = listGroups[4].Id;
+			compCtrl.doSelectItem();
+			system.assertEquals(1     ,compCtrl.getSelectedCount());
+			compCtrl.doDeselectItem();
+			system.assertEquals(0     ,compCtrl.getSelectedCount());
+			
+			compCtrl.last();
+			system.assertEquals(4     ,compCtrl.campGroups.Size());
+			
+			compCtrl.first();
+			system.assertEquals(5     ,compCtrl.campGroups.Size());
+			
+			compCtrl.doNext();
+			system.assertEquals(2     ,compCtrl.getPageNumber());
+			system.assertEquals(true  ,compCtrl.getHasPrevious());
+			system.assertEquals(true  ,compCtrl.getHasNext());
+
+			
+			compCtrl.doPrevious();
+			system.assertEquals(false  ,compCtrl.getHasPrevious());
+			system.assertEquals(true   ,compCtrl.getHasNext());
+
+			compCtrl.editGroups();
+			system.assertEquals(false ,compCtrl.getOutputMode());
+			system.assertEquals(false ,compCtrl.savedByParent);
+			compCtrl.savedByParent =true;
+			system.assertEquals(true  ,compCtrl.getOutputMode());
+			compCtrl.Cancel();
+			system.assertEquals(true  ,compCtrl.getOutputMode());
+	    }
+	}
+	static testMethod void WraperClasssControllerTest() 
```

Comment:
* I suggest to annotate test methods with `@IsTest` for better readability.
* Methods should use the `lowerCamelCase` naming convention. I suggest rename accordingly.